Update documentation about our documentation infrastructure at mystmd.org (#1958)
jupyter-book/mystmd: docs/contribute-docs.md

This will build the documentation locally so that you can preview what your changes will look like.

> ## How update the content at mystmd.org

Reviewer: ```suggestion
## How to update the content at mystmd.org
```

When you change the content here or in most other MyST repositories, it will _not_ automatically update mystmd.org. You must take one extra step, described below.

> The content at mystmd.org is automatically updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.

Reviewer: @rowanc1 can you remind me if this is intentional that we don't deploy automatically? I could set up a web hook to trigger the deploy workflow.

Author: @agoose77 could we make that a follow-up issue? Or do you want to block this PR on that question?

Reviewer: This is _not_ a blocker. I'm just looking for clarity on _why_ we have this policy. We should prioritise getting these docs merged, and revisit if needs be.

Author: I recommend:

- Creating an issue called `Make mystmd.org automatically update any time changes are merged into content repositories`
- Our first action in that issue should be to decide if we want to do it at all, or if something precludes us doing it.
- If we do want to do it, then the next step is to implement whatever webhooks you're talking about.
- If we don't want to do it, we write down a rationale for why not in that issue, and we close it as a record of the conversation+decision so that we don't have this uncertainty in the future

> The content at mystmd.org is automatically updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.

Reviewer 2: ```suggestion
The content at mystmd.org is updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.
```

Too confusing to use "automatically" here, since you just stated that the process is not automatic. This is a different automatic, I get it, but still confusing.

To trigger that action, you can either:

1. **Trigger a workflow dispatch**. Go to [the Vercel action page](https://github.com/jupyter-book/mystmd.org/actions/workflows/deploy.yml). Click on *Run Workflow* and it will run.

> 2. **Push a commit to `main`**. Any new commit to the `main` branch will trigger this action.

Reviewer 2: ```suggestion
2. **Push a commit to `main`** (of [jupyter-book/mystmd.org](https://github.com/jupyter-book/mystmd.org)). Any new commit to the `main` branch will trigger this action.
```

To make clear, again, that it's not a commit to this repo that will do the trick.

> In either case, once the action completes the content at `mystmd.org` will be updated.

Reviewer 2: A bit weird, because the "will be updated" part is already done. So, would consider removing the sentence, or changing to something like "Once the action completes, the content at `mystmd.org` should be up to date."

This will build the documentation locally so that you can preview what your changes will look like.

## How to update the live website at mystmd.org

When you change the content here or in most other MyST repositories, it will _not_ automatically update mystmd.org. This is because we use [a custom MyST theme](#about:mystmd.org) that aggregates content from several MyST repositories into a single website.

> The live website mystmd.org is **managed by a Curvenote deployment**. To update mystmd.org, you must make a deployment to Curvenote.

Reviewer 3: Only the content is hosted on Curvenote, the theme is not.

> Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. We are working on turning this process into a GitHub Action.

Author: ```suggestion
Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. See [this issue about standardizing team access to documentation infrastructure](https://github.com/jupyter-book/team-compass/issues/25) for making this easier for our team to use.
```

> Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. See [this issue about standardizing team access to documentation infrastructure](https://github.com/jupyter-book/team-compass/issues/25) for making this easier for our team to use.

Reviewer 3: Only content hosted on Curvenote, not the theme.

The documentation in this repository serves the content that lives [at `mystmd.org/guide`](https://mystmd.org/guide).

The full content and theme for [`mystmd.org`](https://mystmd.org) is [hosted at `jupyter-book/mystmd.org`](https://github.com/jupyter-book/mystmd.org), which pulls in the content from this folder along with several other documentation sources into one place.

However, the documentation here can be built independently for previewing your changes.

See [](#about:mystmd.org) for more information about the mystmd.org deployment.

(about:mystmd.org)=

## About the `mystmd.org` documentation theme and hosting service

The [MyST website at mystmd.org](https://mystmd.org) is a custom MyST theme designed by the community in order to aggregate documentation from many locations into one website.

### Where is the mystmd.org theme located?

The [jupyter-book/mystmd.org](https://github.com/jupyter-book/mystmd.org) repository has a custom MyST theme that is used to control the content and structure of the site at mystmd.org. It is a [Remix](https://remix.run/) website and is deployed on [Vercel](https://vercel.com/). 

(team-vercel-deployment)=

> ### How to access the Vercel configuration for mystmd.org

Reviewer 3: There is no config, so not sure this section is actually useful?

Currently the theme is run on **Rowan's** personal Vercel project.

> If you need to change the configuration reach out to him.

Reviewer 3: There is no config!

  - [Book theme source](https://github.com/jupyter-book/myst-theme/tree/main/themes/book) -> [Book theme built version](https://github.com/myst-templates/book-theme)
  - [Article theme source](https://github.com/jupyter-book/myst-theme/tree/main/themes/article) -> [Article theme built version](https://github.com/myst-templates/article-theme)

The **content of mystmd.org** is pulled from a number of repositories across the `jupyter-book` organization.

> For example, `mystmd.org/guide` is pulled from [`jupyter-book/mystmd: /docs/`](https://github.com/jupyter-book/mystmd/tree/main/docs), and [`mystmd.org/jtex`](https://mystmd.org/jtex) is pulled from [`jupyter-book/mystmd: /jtex/docs/`](https://github.com/jupyter-book/mystmd/tree/main/packages/jtex/docs). It also includes some custom applications like the Sandbox and MyST demo on the landing page.

Reviewer 2: Good to have these year; a full list would be helpful, but annoying to duplicate. Could we link to the Vercel config, or commit that config to the repo like we do with Netlify?

Reviewer 2: I have never used Vercel, I'm afraid!

Reviewer 3: There is no config. Just select "remix" preset from the dropdown. We aren't doing anything fancy here!

  - [Book theme source](https://github.com/jupyter-book/myst-theme/tree/main/themes/book) -> [Book theme built version](https://github.com/myst-templates/book-theme)
  - [Article theme source](https://github.com/jupyter-book/myst-theme/tree/main/themes/article) -> [Article theme built version](https://github.com/myst-templates/article-theme)

The **content of mystmd.org** is pulled from a number of repositories across the `jupyter-book` organization.

For example, `mystmd.org/guide` is pulled from [`jupyter-book/mystmd: /docs/`](https://github.com/jupyter-book/mystmd/tree/main/docs), and [`mystmd.org/jtex`](https://mystmd.org/jtex) is pulled from [`jupyter-book/mystmd: /jtex/docs/`](https://github.com/jupyter-book/mystmd/tree/main/packages/jtex/docs). It also includes some custom applications like the Sandbox and MyST demo on the landing page.

### How do I deploy changes to the mystmd.org theme?

After updating the theme at `mystmd.org`, we must [deploy it to Vercel](#team-vercel-deployment) in order to make it available for use.

> We use [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.

Reviewer 3: The website doesn't need to be redeployed to get new content live.
